src/components/CollapsableText/CollapsableText.spec.js

```diff
@@ -1,72 +1,74 @@
-import assert from 'assert';
-import { mount } from 'enzyme';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import React from 'react';
-import Button from '../Button/Button';
 import CollapsableText from './CollapsableText';
 
+const mockElement = {
+  clientHeight: 1,
+  scrollHeight: 2,
+};
+
+jest.mock('../../hooks/useOnResizeRef', () => {
+  return {
+    useOnResizeRef: jest.fn((cb) => () => cb(mockElement)),
+  };
+});
+
 describe('<CollapsableText />', () => {
-  it('should show whole string without button if string length is smaller than maxLength', () => {
-    const component = mount(<CollapsableText>Hello World</CollapsableText>);
-    assert.equal(component.text(), 'Hello World');
+  beforeEach(() => {
+    mockElement.scrollHeight = 2;
   });
 
-  it('should shorten the string and show button if string length is greater than maxLength', () => {
-    const component = mount(<CollapsableText maxLength={5}>Hello World</CollapsableText>);
-    assert.equal(component.text(), 'Hello… Show More');
+  it('should show the whole string without button if line length is less than maxLines', () => {
+    mockElement.scrollHeight = 1;
+    render(<CollapsableText maxLines={1}>Hello World</CollapsableText>);
+    screen.getByText('Hello World');
+    expect(screen.queryByText('Show More')).toBeNull();
   });
 
-  it('should show whole string and show button if string length is greater than maxLength and default is to show all', () => {
-    const component = mount(
-      <CollapsableText maxLength={5} collapsed={false}>
-        Hello World
+  it('should limit the container style.maxHeight and show a "Show More" button', () => {
+    render(
+      <CollapsableText maxLines={1}>
+        Hello World <br /> Wow
       </CollapsableText>
     );
-    assert.equal(component.text(), 'Hello World Show Less');
+    expect(screen.getByText(/Hello World/).style.maxHeight).toBeTruthy();
+    screen.getByText('Show More');
```

Comment:
[nit] do you want an assertion here for the 'Show More' button?
`expect(screen.queryByText('Show More')).toBeNull()`

Reply:
`getByText` will throw if the element can't be found. I've gotten used to this style but agree it is more clear to use an assertion.

Reply:
Gotcha, yea this works! 

I only mention because I recalled from kent dodd's tutorial that he [suggests](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions) adding an assertion for readability like you said. 

I've also personally only used `getBy*` to insure that an element critical to the functionality of the test is present (e.g. get 'Show More' button to click and verify the full text is displayed) and `queryBy*` for assertions, but tbh I have to familiarize myself with how y'all use react testing library.